pallets/nfts/src/features/approvals.rs

```diff
@@ -165,10 +181,191 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		owner: T::AccountId,
+		collection: T::CollectionId,
+		limit: u32,
+	) -> Result<u32, DispatchError> {
+		if limit == 0 {
+			return Ok(0);
+		}
+		let mut removed_approvals: u32 = 0;
+		let mut deposits: BalanceOf<T, I> = Zero::zero();
+		// Iterate and remove each collection approval, returning the deposit back to the `owner`.
+		for (_, (_, deposit)) in CollectionApprovals::<T, I>::drain_prefix((collection, &owner)) {
+			deposits = deposits.saturating_add(deposit);
+			removed_approvals.saturating_inc();
+			if removed_approvals == limit {
+				break;
+			}
+		}
+
+		T::Currency::unreserve(&owner, deposits);
+		Self::deposit_event(Event::ApprovalsCancelled {
+			collection,
+			item: None,
+			owner,
+			approvals: removed_approvals,
+		});
+		Ok(removed_approvals)
+	}
+
+	/// Checks whether the `delegate` is approved to transfer collection items of `owner`.
+	///
+	/// - `collection`: The identifier of the collection.
+	/// - `owner`: The owner of the collection items.
+	/// - `delegate`: The account to check for approval to transfer collection items of `owner`.
+	fn check_collection_approval(
+		collection: &T::CollectionId,
+		owner: &T::AccountId,
+		delegate: &T::AccountId,
+	) -> DispatchResult {
+		let (maybe_deadline, _) =
+			CollectionApprovals::<T, I>::get((&collection, &owner, &delegate))
+				.ok_or(Error::<T, I>::NoPermission)?;
+		if let Some(deadline) = maybe_deadline {
+			let block_number = frame_system::Pallet::<T>::block_number();
+			ensure!(block_number <= deadline, Error::<T, I>::ApprovalExpired);
+		}
+		Ok(())
+	}
+
+	/// Checks whether the `delegate` is approved by `owner` to transfer its collection item(s). If
+	/// the `delegate` is approved for all `owner`'s collection items it can transfer every item
+	/// without requiring explicit approval for an item.
+	///
+	/// - `collection`: The identifier of the collection.
+	/// - `maybe_item`: The optional item of the collection that the delegated account has an
+	///   approval to transfer. If not provided, an approval to transfer all `owner`'s collection
+	///   items will be checked.
+	/// - `owner`: The owner of the specified collection item.
+	/// - `delegate`: The account to check for approval to transfer collection item(s) of owner.
+	pub fn check_approval(
+		collection: &T::CollectionId,
+		maybe_item: &Option<T::ItemId>,
+		owner: &T::AccountId,
+		delegate: &T::AccountId,
+	) -> DispatchResult {
+		// Check if `delegate` has permission to transfer `owner`'s collection items.
+		let Err(error) = Self::check_collection_approval(collection, owner, delegate) else {
+			return Ok(());
+		};
+
+		// Check if a `delegate` has permission to transfer the given collection item.
+		if let Some(item) = maybe_item {
+			let details = Item::<T, I>::get(collection, item).ok_or(Error::<T, I>::UnknownItem)?;
+			let maybe_deadline =
+				details.approvals.get(delegate).ok_or(Error::<T, I>::NoPermission)?;
+			if let Some(deadline) = maybe_deadline {
```

Comment:
No action required, but this same code exists in the function above.

Reply:
I think it will be cleaner if both functions share the `is_deadline_valid` method so I resolved by adding it.